client/ping.go

```diff
@@ -71,6 +74,42 @@ func parsePingResponse(cli *Client, resp serverResponse) (types.Ping, error) {
 			ControlAvailable: role == "manager",
 		}
 	}
+	if engineFeatures := resp.header.Get("Engine-Features"); engineFeatures != "" {
```

Comment:
One thing here is that this only uses the first `Engine-Features` header. If we want to support multiple, then this should use `values := resp.header.Values("Engine-Features")` and iterate over those values (splitting each), but in that case we should take the total items into account (which may be split over multiple headers).

Mostly considering here that _iff_ Docker Desktop would want to be adding its own features, it may be easier for there to just set an additional header than to parse the header and append value(s).

One thing I didn't consider in all of those scenarios is what to do with possible duplicates (`feature=true,feature=false`)

Reply:
> One thing I didn't consider in all of those scenarios is what to do with possible duplicates (feature=true,feature=false)

IMO the only sensible behavior here is to have the client error out. There's no possible way for the client to know what it should do in that case, and we shouldn't try to be smart about this.

Reply:
Added handling for duplicate features ✅